src/components/Button/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState, type FC, type HTMLAttributes, type PropsWithChildren } from 'react'
+
+import { IconProp } from '@fortawesome/fontawesome-svg-core'
+import { Icon } from '../Icon'
+import * as S from './Button.styles'
+import type { StyleProps } from './Button.types'
+
+interface Props extends HTMLAttributes<HTMLButtonElement>, StyleProps {
+  icon?: IconProp
+  onClick?: () => void
+}
+
+export const Button: FC<Props> = ({ children, icon, onClick, ...props }: PropsWithChildren<Props>) => {
+  const [isClicked, setIsClicked] = useState(false)
+  const handleClick = () => {
+    setIsClicked(true)
+    setTimeout(() => {
+      setIsClicked(false)
+    }, 150)
+    if (onClick) {
+      onClick()
+    }
+  }
+  return (
+    <S.Button {...props} onClick={handleClick} className={isClicked ? 'clicked' : ''}>
```

Comment:
```suggestion
    <S.Button {...props} onClick={handleClick} isClicked={isClicked}>
```
위와 같이해서 true/false일 때 스타일을 다르게 해도 될 것 같습니다.

Reply:
넵, 이해했습니다. 수정하겠습니다.!